app/helpers/application_helper/button/middleware_instance_add.rb

```diff
@@ -0,0 +1,13 @@
+class ApplicationHelper::Button::MiddlewareInstanceAdd < ApplicationHelper::Button::MiddlewareAction
```

Comment:
Thank you for straightening that one out. I had big issues naming the classes as I did not know the common features of the refactored buttons. I chose `MiddlewareInstanceAdd` since there were only `*_add` buttons first for that case. You extended the case later.

I like your suggestion and decided to rename the classes to `MiddlewareServerAction` and `MiddlewareStandAloneServerAction`. However, I think that a better solution would be to give the button classes a logical structure, hence creating subfolders/modules in the form of `ApplicationHelper::Button::Middleware::ServerAction`. @romanblanco do you agree?